.github/workflows/test.yml

```
@@ -28,12 +28,33 @@ jobs:
       - run: go build -v .
       - name: Run linters
         uses: golangci/golangci-lint-action@971e284b6050e8a5849b72094c50ab08da042db8 # v6.1.1
+  
+  configure-aws-credentials:
+    name: Configure AWS Credentials
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    outputs:
+      LAUNCHDARKLY_ACCESS_TOKEN: ${{ steps.get-launchdarkly-access-token.outputs.LAUNCHDARKLY_ACCESS_TOKEN }}
+    steps:
+      - uses: aws-actions/configure-aws-credentials@f24d7193d98baebaeacc7e2227925dd47cc267f5 # v4.2.0
+        with:
+          audience: https://github.com/launchdarkly
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::061661829416:role/github-actions-terraform-provider-launchdarkly
+      - id: get-launchdarkly-access-token
+        uses: dkershner6/aws-ssm-getparameters-action@4fcb4872421f387a6c43058473acc1b22443fe13 # v2.0.3
+        with:
+          parameterPairs: |
+            /global/services/github/terraform-provider/launchdarkly-access-token = LAUNCHDARKLY_ACCESS_TOKEN
 
   generate:
+    needs: configure-aws-credentials
     runs-on: ubuntu-latest
     timeout-minutes: 5
     env:
-      LAUNCHDARKLY_ACCESS_TOKEN: ${{ secrets.LAUNCHDARKLY_ACCESS_TOKEN }}
+      LAUNCHDARKLY_ACCESS_TOKEN: ${{ needs.configure-aws-credentials.outputs.LAUNCHDARKLY_ACCESS_TOKEN || secrets.LAUNCHDARKLY_ACCESS_TOKEN }}
```

I'd probably remove the `OR` from this given it'll always fetch the access token from Parameter Store
```suggestion
      LAUNCHDARKLY_ACCESS_TOKEN: ${{ needs.configure-aws-credentials.outputs.LAUNCHDARKLY_ACCESS_TOKEN }}
```